### PR TITLE
Support stencil buffer when post-processing is disabled

### PR DESCRIPTION
**Pull request by bejado**
Still need to add support for other backends' SwapChain implementations.

- [ ] OpenGL support
- [ ] Vulkan support

To test, in gltf_viewer, turn on Debug overdraw and turn off post-processing.

### CODE REVIEW

File: filament/src/details/Renderer.cpp [hunk 1]
```diff
@@ -1095,6 +1095,13 @@ void FRenderer::renderJob(ArenaScope& arena, FView& view) {
         }
     }
 
+    if (outputIsSwapChain && view.isStencilBufferEnabled()) {
```
Comment (pixelflinger) [inline filament/src/details/Renderer.cpp:1098]:
```suggestion
    if (UTILS_UNLIKELY(outputIsSwapChain && view.isStencilBufferEnabled())) {
```

File: filament/src/details/Renderer.cpp [hunk 2]
```diff
@@ -1095,6 +1095,13 @@ void FRenderer::renderJob(ArenaScope& arena, FView& view) {
...
+        assert_invariant(mSwapChain);
+        ASSERT_PRECONDITION(mSwapChain->hasStencilBuffer(),
+                "View has stencil buffer enabled, but SwapChain does not have "
+                "CONFIG::CONFIG_HAS_STENCIL_BUFFER flag set.");
```
Comment (pixelflinger) [inline filament/src/details/Renderer.cpp:1102]:
```suggestion
                "SwapChain::CONFIG_HAS_STENCIL_BUFFER flag set.");
```

File: NEW_RELEASE_NOTES.md [hunk 3]
```diff
@@ -9,4 +9,5 @@ appropriate header in [RELEASE_NOTES.md](./RELEASE_NOTES.md).
 ## Release notes for next branch cut
 
 engine: Added parameter for configuring JobSystem thread count
-engine: In Java, decpreate Engine.create() and recommend Engine.Builder instead
\ No newline at end of file
+engine: In Java, decpreate Engine.create() and recommend Engine.Builder instead
+engine: Add support for stencil buffer when post-processing is disabled.
```
Comment (pixelflinger) [inline NEW_RELEASE_NOTES.md:13]:
```suggestion
engine: Add support for stencil buffer when post-processing is disabled (Metal backend only).
```

**Review by pixelflinger — CHANGES_REQUESTED** [review]
Please file an issue to implement support for gl and vk.

**Review by poweifeng — APPROVED** [review]

**Pull request merged**